zingolib/src/wallet/transaction_records_by_id.rs

```diff
@@ -507,4 +514,31 @@ mod tests {
 
         assert_eq!(transaction_records_by_id.len(), 1);
     }
+
+    #[test]
+    fn get_received_note_from_identifier() {
+        let mut trbid = TransactionRecordsById::new();
+        trbid.insert_transaction_record(setup_mock_transaction_record());
+
+        let received_note = trbid.get_received_note_from_identifier::<SaplingDomain>(
+            crate::wallet::notes::ShNoteId {
+                txid: default_txid(),
+                shpool: ShieldedProtocol::Sapling,
+                index: 0,
+            },
+        );
+
```

Comment:
can we expand this test to prove the correct note is selected between 2 options?